app/src/org/commcare/dalvik/dialogs/AlertDialogFragment.java

```diff
@@ -32,6 +34,11 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
+    public void onCancel(DialogInterface dialog) {
+        factory.performCancel(dialog);
```

Comment:
I think this is probably fine and correct, I'm not sure what else you would do besides overriding the onCancel() here.